packages/replayio/src/utils/async/timeoutAfter.ts

```diff
@@ -0,0 +1,15 @@
+type TimeoutResult = { timedOutAfter: number };
+
+export async function timeoutAfter(duration: number): Promise<TimeoutResult> {
+  const startTime = Date.now();
+  return new Promise(resolve =>
+    setTimeout(() => {
+      const endTime = Date.now();
+      resolve({ timedOutAfter: endTime - startTime });
```

Comment:
Yeah.

The reason I added this _shape_ was because `raceWithTimeout` needs to be able to differentiate between a timed-out promise (`isTimeoutResult`) and the thing it's racing.

The reason I added the _actual_ duration vs the passed one 🤷🏼  I guess it just felt slightly more correct.